### PR TITLE
Add metadata needed for improved documentation generation

### PR DESCRIPTION
**Pull request by cgranade**
This PR adds package names to documentation metadata, adapts C# magic command implementations to resolve the ambiguity between `Documentation` from the Microsoft.Quantum.* and Microsoft.Jupyter.Core namespaces, and adds `/// # Named Items` sections. See https://github.com/microsoft/qsharp-compiler/pull/594 for additional context and discussion.

### CODE REVIEW

File: Numerics/src/Numerics.csproj [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
     <AssemblyName>Microsoft.Quantum.Numerics</AssemblyName>
     <QsharpDocsGeneration>true</QsharpDocsGeneration>
+    <QsharpDocsPackageId>Microsoft.Quantum.Numerics</QsharpDocsPackageId>
     <IncludeQsharpCorePackages>false</IncludeQsharpCorePackages>
   </PropertyGroup>
 
   <PropertyGroup>
     <NoWarn>0162</NoWarn>
-    <RunQDocGen>True</RunQDocGen>
+    <RunQDocGen>True</RunQDocGen>https://github.com/microsoft/qsharp-runtime/pull/390
```
Comment (msoeken) [inline Numerics/src/Numerics.csproj:13]: That does not look like valid XML

**Review by msoeken — APPROVED** [review]
Thanks @cgranade. One change looked a bit odd, and I commented it. Otherwise good to go.

File: Numerics/src/Numerics.csproj (hunk 1 above)
Comment (cgranade) [reply Numerics/src/Numerics.csproj:13]: It's not at all, no. I'll blame a copy-paste error. Thanks for catching it!
```suggestion
    <RunQDocGen>True</RunQDocGen>
```

**Pull request merged**